src/http/request.cr

```diff
@@ -95,11 +95,14 @@ class HTTP::Request
     return BadRequest.new unless parts.size == 3
 
     method, resource, http_version = parts
-    HTTP.parse_headers_and_body(io) do |headers, body|
-      return new method, resource, headers, body, http_version
+    begin
+      HTTP.parse_headers_and_body(io) do |headers, body|
+        return new method, resource, headers, body, http_version
+      end
+    rescue ex : ArgumentError
```

Comment:
I'm not entirely sure if this is the correct way to detect this. It feels to broad. I'm not sure what would be a better approach though.

Reply:
@RX14 To clarify: are you thinking about more exceptions which could be thrown and would be handled like this?

Reply:
Perhaps non-`ArgumentError` exceptions would be thrown, perhaps `new` would throw an exception. This PR is a great improvement but perhaps we could implement it a better way.

Reply:
The best way probably would be not rescuing the exception but handling the case that triggers it explicitly/differently.

Reply:
@jhass @RX14 updated, now it is handled explicitly in `parse_headers_and_body` while adding header. Ideas?

Reply:
Ideally no exception should be raised nor rescued. We should maybe have a method to add a header and return `false` or `nil` if it's not valid, instead of raising.

Reply:
Also some tests would be awesome :)